File: 백준/Bronze/31868. 수박 게임/수박 게임.java
import java.util.Scanner;


public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int K = sc.nextInt();
        int cnt = 1;
        while (true) {
            if (cnt == N){
                System.out.println(K);
                break;
            }
            K/=2;
            cnt+=1;
        }
    }
    

}

File: 백준/Bronze/31868. 수박 게임/수박 게임.c
#include <stdio.h>

int main(void){
  int N, K;
  scanf("%d %d", &N, &K);
  int cnt = 1;
  while (1) {
    if (cnt == N) {
      printf("%d", K);
      break;
    }
    K /= 2;
    cnt += 1;
  }
  return 0;
}